utf8cpp's test suite fails on darwin

### Describe the bug

Current `master`:

```
unpacking sources
unpacking source archive /nix/store/rjm0v5s7bsyvbl6d0b3vyb460f197f25-source
source root is source
patching sources
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_LOCALEDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/share/doc/utf8cpp -DCMAKE_INSTALL_INFODIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/5j78r51lhl6pwfjxd2cz25p40xcnnywn-cctools-binutils-darwin-949.0.1/bin/strip -DCMAKE_RANLIB=/nix/store/5j78r51lhl6pwfjxd2cz25p40xcnnywn-cctools-binutils-darwin-949.0.1/bin/ranlib -DCMAKE_AR=/nix/store/5j78r51lhl6pwfjxd2cz25p40xcnnywn-cctools-binutils-darwin-949.0.1/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/rfnafncznyd5z6rjjqw008mx547061m0-utf8cpp-3.2.1 -DCMAKE_INSTALL_LIBDIR=lib 
-- The CXX compiler identification is Clang 7.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/mc3wc45mhv836fs8dnjw4rfm7pj7gljl-clang-wrapper-7.1.0/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_FIND_USE_PACKAGE_REGISTRY
    CMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/build
cmake: enabled parallel building
building
build flags: -j4 -l4 SHELL=/nix/store/vm564n0g8k6pkiwqb31x1dw1y02h2mak-bash-5.1-p8/bin/bash
[  6%] Building CXX object CMakeFiles/docsample.dir/samples/docsample.cpp.o
[ 18%] Building CXX object tests/CMakeFiles/cpp17.dir/test_cpp17.cpp.o
[ 25%] Building CXX object tests/CMakeFiles/apitests.dir/test_checked_api.cpp.o
[ 18%] Building CXX object tests/CMakeFiles/negative.dir/negative.cpp.o
In file included from /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/test_checked_api.cpp:2:
/tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/../extern/ftest/ftest.h:159:13: warning: delete called on 'ftest::Test' that is abstract but has non-virtual destructor [-Wdelete-non-virtual-dtor]
            delete m_tests[i];
            ^
In file included from /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/test_cpp17.cpp:1:
/tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/../extern/ftest/ftest.h:159:13: warning: delete called on 'ftest::Test' that is abstract but has non-virtual destructor [-Wdelete-non-virtual-dtor]
            delete m_tests[i];
            ^
[ 31%] Linking CXX executable negative
[ 31%] Built target negative
[ 37%] Building CXX object tests/CMakeFiles/noexceptionstests.dir/test_unchecked_api.cpp.o
1 warning generated.
[ 43%] Building CXX object tests/CMakeFiles/apitests.dir/test_unchecked_api.cpp.o
1 warning generated.
[ 50%] Linking CXX executable cpp17
[ 56%] Linking CXX executable docsample
[ 56%] Built target cpp17
[ 62%] Building CXX object tests/CMakeFiles/apitests.dir/test_checked_iterator.cpp.o
[ 62%] Built target docsample
[ 68%] Building CXX object tests/CMakeFiles/apitests.dir/test_unchecked_iterator.cpp.o
In file included from /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/test_unchecked_api.cpp:1:
/tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/../extern/ftest/ftest.h:159:13: warning: delete called on 'ftest::Test' that is abstract but has non-virtual destructor [-Wdelete-non-virtual-dtor]
            delete m_tests[i];
            ^
In file included from /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/test_unchecked_api.cpp:1:
/tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/../extern/ftest/ftest.h:159:13: warning: delete called on 'ftest::Test' that is abstract but has non-virtual destructor [-Wdelete-non-virtual-dtor]
            delete m_tests[i];
            ^
In file included from /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/test_unchecked_iterator.cpp:2:
/tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/../extern/ftest/ftest.h:159:13: warning: delete called on 'ftest::Test' that is abstract but has non-virtual destructor [-Wdelete-non-virtual-dtor]
            delete m_tests[i];
            ^
In file included from /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/test_checked_iterator.cpp:2:
/tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/../extern/ftest/ftest.h:159:13: warning: delete called on 'ftest::Test' that is abstract but has non-virtual destructor [-Wdelete-non-virtual-dtor]
            delete m_tests[i];
            ^
1 warning generated.
[ 75%] Building CXX object tests/CMakeFiles/noexceptionstests.dir/test_unchecked_iterator.cpp.o
1 warning generated.
1 warning generated.
[ 81%] Building CXX object tests/CMakeFiles/cpp11.dir/test_cpp11.cpp.o
1 warning generated.
[ 87%] Linking CXX executable apitests
In file included from /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/test_unchecked_iterator.cpp:2:
/tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/../extern/ftest/ftest.h:159:13: warning: delete called on 'ftest::Test' that is abstract but has non-virtual destructor [-Wdelete-non-virtual-dtor]
            delete m_tests[i];
            ^
[ 87%] Built target apitests
In file included from /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/test_cpp11.cpp:1:
/tmp/nix-build-utf8cpp-3.2.1.drv-0/source/tests/../extern/ftest/ftest.h:159:13: warning: delete called on 'ftest::Test' that is abstract but has non-virtual destructor [-Wdelete-non-virtual-dtor]
            delete m_tests[i];
            ^
1 warning generated.
[ 93%] Linking CXX executable noexceptionstests
[ 93%] Built target noexceptionstests
1 warning generated.
[100%] Linking CXX executable cpp11
[100%] Built target cpp11
running tests
check flags: SHELL=/nix/store/vm564n0g8k6pkiwqb31x1dw1y02h2mak-bash-5.1-p8/bin/bash VERBOSE=y test
Running tests...
/nix/store/bn00vnhjd5p1ry7j1ipfakfnnqs1bbhs-cmake-3.21.2/bin/ctest --force-new-ctest-process 
Test project /tmp/nix-build-utf8cpp-3.2.1.drv-0/source/build
    Start 1: negative_test
    Start 2: cpp11_test
    Start 3: cpp17_test
    Start 4: api_test
1/5 Test #1: negative_test ....................   Passed    0.01 sec
    Start 5: noexceptions_test
2/5 Test #2: cpp11_test .......................***Exception: Illegal  0.04 sec
[==========] Running 9 tests from 1 test cases.
[----------] 9 tests from CPP11APITests
[ RUN      ] CPP11APITests.test_append
[       OK ] CPP11APITests.test_append
[ RUN      ] CPP11APITests.test_utf16to8
[       OK ] CPP11APITests.test_utf16to8
[ RUN      ] CPP11APITests.test_utf8to16
[       OK ] CPP11APITests.test_utf8to16
[ RUN      ] CPP11APITests.test_utf32to8
[       OK ] CPP11APITests.test_utf32to8
[ RUN      ] CPP11APITests.test_utf8to32
[       OK ] CPP11APITests.test_utf8to32
[ RUN      ] CPP11APITests.test_find_invalid
[       OK ] CPP11APITests.test_find_invalid
[ RUN      ] CPP11APITests.test_is_valid
[       OK ] CPP11APITests.test_is_valid
[ RUN      ] CPP11APITests.test_replace_invalid
[       OK ] CPP11APITests.test_replace_invalid
[ RUN      ] CPP11APITests.test_starts_with_bom
[       OK ] CPP11APITests.test_starts_with_bom
[----------] 9 tests from CPP11APITests
[==========] 9 tests from 1 test cases ran.
[  PASSED  ] 9 tests.

3/5 Test #3: cpp17_test .......................***Exception: Illegal  0.14 sec
[==========] Running 9 tests from 1 test cases.
[----------] 9 tests from CPP17APITests
[ RUN      ] CPP17APITests.test_utf16to8
[       OK ] CPP17APITests.test_utf16to8
[ RUN      ] CPP17APITests.test_utf8to16
[       OK ] CPP17APITests.test_utf8to16
[ RUN      ] CPP17APITests.test_utf32to8
[       OK ] CPP17APITests.test_utf32to8
[ RUN      ] CPP17APITests.test_utf8to32
[       OK ] CPP17APITests.test_utf8to32
[ RUN      ] CPP17APITests.test_find_invalid
[       OK ] CPP17APITests.test_find_invalid
[ RUN      ] CPP17APITests.test_is_valid
[       OK ] CPP17APITests.test_is_valid
[ RUN      ] CPP17APITests.test_replace_invalid
[       OK ] CPP17APITests.test_replace_invalid
[ RUN      ] CPP17APITests.test_starts_with_bom
[       OK ] CPP17APITests.test_starts_with_bom
[ RUN      ] CPP17APITests.string_class_and_literals
[       OK ] CPP17APITests.string_class_and_literals
[----------] 9 tests from CPP17APITests
[==========] 9 tests from 1 test cases ran.
[  PASSED  ] 9 tests.

4/5 Test #4: api_test .........................***Exception: Illegal  0.14 sec
[==========] Running 15 tests from 3 test cases.
[----------] 11 tests from UnCheckedAPITests
[ RUN      ] UnCheckedAPITests.test_append
[       OK ] UnCheckedAPITests.test_append
[ RUN      ] UnCheckedAPITests.test_next
[       OK ] UnCheckedAPITests.test_next
[ RUN      ] UnCheckedAPITests.test_peek_next
[       OK ] UnCheckedAPITests.test_peek_next
[ RUN      ] UnCheckedAPITests.test_prior
[       OK ] UnCheckedAPITests.test_prior
[ RUN      ] UnCheckedAPITests.test_advance
[       OK ] UnCheckedAPITests.test_advance
[ RUN      ] UnCheckedAPITests.test_distance
[       OK ] UnCheckedAPITests.test_distance
[ RUN      ] UnCheckedAPITests.test_utf32to8
[       OK ] UnCheckedAPITests.test_utf32to8
[ RUN      ] UnCheckedAPITests.test_utf8to32
[       OK ] UnCheckedAPITests.test_utf8to32
[ RUN      ] UnCheckedAPITests.test_utf16to8
[       OK ] UnCheckedAPITests.test_utf16to8
[ RUN      ] UnCheckedAPITests.test_utf8to16
[       OK ] UnCheckedAPITests.test_utf8to16
[ RUN      ] UnCheckedAPITests.test_replace_invalid
[       OK ] UnCheckedAPITests.test_replace_invalid
[----------] 11 tests from UnCheckedAPITests
[----------] 2 tests from CheckedIteratrTests
[ RUN      ] CheckedIteratrTests.test_increment
[       OK ] CheckedIteratrTests.test_increment
[ RUN      ] CheckedIteratrTests.test_decrement
[       OK ] CheckedIteratrTests.test_decrement
[----------] 2 tests from CheckedIteratrTests
[----------] 2 tests from UnCheckedIteratrTests
[ RUN      ] UnCheckedIteratrTests.test_increment
[       OK ] UnCheckedIteratrTests.test_increment
[ RUN      ] UnCheckedIteratrTests.test_decrement
[       OK ] UnCheckedIteratrTests.test_decrement
[----------] 2 tests from UnCheckedIteratrTests
[==========] 15 tests from 3 test cases ran.
[  PASSED  ] 15 tests.

5/5 Test #5: noexceptions_test ................***Exception: Illegal  0.18 sec
[==========] Running 13 tests from 2 test cases.
[----------] 11 tests from UnCheckedAPITests
[ RUN      ] UnCheckedAPITests.test_append
[       OK ] UnCheckedAPITests.test_append
[ RUN      ] UnCheckedAPITests.test_next
[       OK ] UnCheckedAPITests.test_next
[ RUN      ] UnCheckedAPITests.test_peek_next
[       OK ] UnCheckedAPITests.test_peek_next
[ RUN      ] UnCheckedAPITests.test_prior
[       OK ] UnCheckedAPITests.test_prior
[ RUN      ] UnCheckedAPITests.test_advance
[       OK ] UnCheckedAPITests.test_advance
[ RUN      ] UnCheckedAPITests.test_distance
[       OK ] UnCheckedAPITests.test_distance
[ RUN      ] UnCheckedAPITests.test_utf32to8
[       OK ] UnCheckedAPITests.test_utf32to8
[ RUN      ] UnCheckedAPITests.test_utf8to32
[       OK ] UnCheckedAPITests.test_utf8to32
[ RUN      ] UnCheckedAPITests.test_utf16to8
[       OK ] UnCheckedAPITests.test_utf16to8
[ RUN      ] UnCheckedAPITests.test_utf8to16
[       OK ] UnCheckedAPITests.test_utf8to16
[ RUN      ] UnCheckedAPITests.test_replace_invalid
[       OK ] UnCheckedAPITests.test_replace_invalid
[----------] 11 tests from UnCheckedAPITests
[----------] 2 tests from UnCheckedIteratrTests
[ RUN      ] UnCheckedIteratrTests.test_increment
[       OK ] UnCheckedIteratrTests.test_increment
[ RUN      ] UnCheckedIteratrTests.test_decrement
[       OK ] UnCheckedIteratrTests.test_decrement
[----------] 2 tests from UnCheckedIteratrTests
[==========] 13 tests from 2 test cases ran.
[  PASSED  ] 13 tests.


20% tests passed, 4 tests failed out of 5

Total Test time (real) =   0.22 sec

The following tests FAILED:
          2 - cpp11_test (ILLEGAL)
          3 - cpp17_test (ILLEGAL)
          4 - api_test (ILLEGAL)
          5 - noexceptions_test (ILLEGAL)
Errors while running CTest
make: *** [Makefile:136: test] Error 8
```


### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

cc @jobojeha

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: utf8cpp
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/libraries/utf8cpp/default.nix
+++ b/pkgs/development/libraries/utf8cpp/default.nix
@@ -18,7 +18,9 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
-  doCheck = true;
+  # Tests fail on darwin, probably due to a bug in the test framework:
+  # https://github.com/nemtrif/utfcpp/issues/84
+  doCheck = !stdenv.isDarwin;
 
   meta = with lib; {
     homepage = "https://github.com/nemtrif/utfcpp";

